### PR TITLE
Codespell: fixed typos, added config and workflow

### PR DESCRIPTION
**Pull request by yarikoptic**
## Motivation

Still on the mission

## How to test the behavior?

Run codespell

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [ ] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

File: docs/gallery/general/extensions.py [hunk 1]
```diff
@@ -326,7 +326,7 @@ class PotatoSack(MultiContainerInterface):
 from dateutil.tz import tzlocal
 
 # You can add potatoes to a potato sack in different ways
-potato_sack = PotatoSack(potatos=Potato(name='potato1', age=2.3, weight=3.0))
+potato_sack = PotatoSack(potatoes=Potato(name='potato1', age=2.3, weight=3.0))
```
Comment (yarikoptic) [inline docs/gallery/general/extensions.py]: just now mentioned this one -- should I also fix `potatos` above or leave any `potatos` untouched?

Comment (rly) [reply docs/gallery/general/extensions.py]: Please leave `potatos` untouched. The variable name is autogenerated, which doesn't apply 100% of the time...

Comment (yarikoptic) [reply docs/gallery/general/extensions.py]: done, left potatos behind.

**Comment by rly** [comment]
Thanks @yarikoptic . Could you please update `CHANGELOG.md`? Otherwise it looks good to me.

**Comment by bendichter** [comment]
@rly added

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # PyNWB Changelog
 
+
+## PyNWB 2.3.2 (Upcoming)
+
+### Enhancements and minor changes
+- Fixed typos and added codespell GitHub action to check spelling in the future. @yarikoptic [#1648](https://github.com/NeurodataWithoutBorders/pynwb/pull/1648)
```
Comment (yarikoptic) [inline CHANGELOG.md:7]: Thank you @bendichter !

**Review by rly — APPROVED** [review]

**Comment by rly** [comment]
Thanks to you both!

**Review by oruebel — APPROVED** [review]

**Pull request merged**